code-challenges.test.js

```diff
@@ -14,32 +14,85 @@
+    let colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+    it("takes in an array, removes the first item from the array and shuffles the remaining content", () => {
+        expect(rmvShuffle(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+        expect(rmvShuffle(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+    })
+})
+
+// RED
 
 // b) Create the function that makes the test pass.
 
+// const rmvShuffle = (array) => {
+//     array.shift()
+//     return array
+// }
+
+// It passed but I'm Not sure if the resulting array is actually shuffled or if the test is passing solely because it contains the items. My knowledge of js tells me that .shift has removed the first item and nothing else because it did pass, will log and run to see.
+// console.log(rmvShuffle(colors2));
+// It didn't shuffle the array as I thought *false GREEN*
+
+
+
+const rmvShuffle = (array) => {
+    array.shift()
+    // create a for loop that randomly removes items from the array and then pushes them onto the end of it
+    for (let index = 0; index < array.length * 3; index++) {
+        let removedValue = array.splice((Math.floor(Math.random() * array.length)), 1)
+        // since .splice returns an array we will push the item from removedValue array at index 0
+        array.push(removedValue[0])
+    }
+    return array
+}
+
+// The real Green shows itsself
+// But I also went down a rabbit hole of shuffling things and generating things that are equally random versus biased and blah blah blah
 
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
-var nums2 = [109, 5, 9, 67, 8, 24]
+// var nums1 = [3, 56, 90, -8, 0, 23, 6]
+// // Expected output: [-8, 90]
+// var nums2 = [109, 5, 9, 67, 8, 24]
 // Expected output: [5, 109]
 
+// describe a function minMax that takes an array of numbers and returns an array of the minimum and maximum numbers in that order
+describe("minMax", () => {
+    let nums1 = [3, 56, 90, -8, 0, 23, 6]
+    let nums2 = [109, 5, 9, 67, 8, 24]
+    it("takes an array of numbers and returns an array of the minimum and maximum numbers in that order", () => {
+        expect(minMax(nums1)).toEqual([-8, 90])
+        expect(minMax(nums2)).toEqual([5, 109])
+ })
+})
 
+// RED
 
 // b) Create the function that makes the test pass.
+// Create a function minMax that takes an array of numbers and returns an array of the minimum and maximum numbers in that order
 
+const minMax = (array) => {
+    // create an array for the min and max values to be pushed into
+    let minMaxArray = []
+    minMaxArray.push(Math.min(...array), Math.max(...array))
+    return minMaxArray
+}
 
+//GREEN
```

Comment:
Nice job. I think you could shorten this even more by just putting your min/max logic in the array. `return [Math.min(...array), Math.max(...array)]`